src/Xenial.Tasty/Execution/TestExecutor.cs

```diff
@@ -1,27 +1,44 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 
+using Xenial.Delicious.Commands;
 using Xenial.Delicious.Execution.TestGroupMiddleware;
 using Xenial.Delicious.Execution.TestMiddleware;
-using Xenial.Delicious.Metadata;
+using Xenial.Delicious.Execution.TestRuntime;
 using Xenial.Delicious.Scopes;
-using Xenial.Delicious.Visitors;
 
 namespace Xenial.Delicious.Execution
 {
     public class TestExecutor
     {
         private IList<Func<TestDelegate, TestDelegate>> TestMiddlewares = new List<Func<TestDelegate, TestDelegate>>();
         private IList<Func<TestGroupDelegate, TestGroupDelegate>> TestGroupMiddlewares = new List<Func<TestGroupDelegate, TestGroupDelegate>>();
+        private IList<Func<RuntimeDelegate, RuntimeDelegate>> RuntimeMiddlewares = new List<Func<RuntimeDelegate, RuntimeDelegate>>();
```

Comment:
I think we can make a generic middleware collection type that ease the use of this pattern?

Reply:
If we need the next middleware we should address this